static/js/components/Source.jsx

```diff
@@ -63,6 +70,63 @@ const Source = ({ route }) => {
     );
   }
 
+  const groupBy = (array, key) => array.reduce((result, cv) => {
+    (result[cv[key]] = result[cv[key]] || []).push(cv);
```

Comment:
This is a bit dense; perhaps an inline comment explaining the logic would help

Reply:
done.